runtime/compiler/runtime/JITServerAOTCache.hpp

```diff
@@ -126,23 +142,21 @@ class AOTCacheClassLoaderRecord final : public AOTCacheRecord
 
    static AOTCacheClassLoaderRecord *create(uintptr_t id, const uint8_t *name, size_t nameLength);
 
-   static AOTCacheClassLoaderRecord *read(FILE *f,
-                                          const Vector<AOTCacheClassLoaderRecord *> &classLoaderRecords,
-                                          const Vector<AOTCacheClassRecord *> &classRecords,
-                                          const Vector<AOTCacheMethodRecord *> &methodRecords,
-                                          const Vector<AOTCacheClassChainRecord *> &classChainRecords,
-                                          const Vector<AOTCacheWellKnownClassesRecord *> &wellKnownClassesRecords,
-                                          const Vector<AOTCacheAOTHeaderRecord *> &aotHeaderRecords);
-
 private:
+   using SerializationRecord = ClassLoaderSerializationRecord;
+
+   template<class R> friend R *AOTCacheRecord::readRecord(JITServerAOTCacheReadContext &context);
```

Comment:
Can `friend` the template specialization for this class rather than the whole template.